obsplus/waveforms/get_waveforms.py

```diff
@@ -56,14 +53,15 @@ def get_waveforms(
     Stream
     """
     stream = stream.copy()
+    time1, time2 = UTC(starttime or SMALL_UTC), UTC(endtime or BIG_UTC)
     st = stream.select(
         network=network, station=station, location=location, channel=channel
     )
-    st = st.trim(starttime=UTC(starttime or SMALL_UTC), endtime=UTC(endtime or BIG_UTC))
+    st.trim(starttime=time1, endtime=time2)
```

Comment:
Have you considered doing `.slice(...).copy()` rather than copying the full stream and trimming - this would probably be more memory efficient. Both `.select` and `.slice` return references to the original Traces, but `.copy` on the sliced trace will return a new stream without affecting the users input AFAIK.

Reply:
Good idea.